designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/BasicInputImage.kt

```diff
@@ -0,0 +1,153 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Cancel
+import androidx.compose.material.icons.outlined.Draw
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.testTag
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+
+/**
+ * DHIS2 BasicInputImage. Wraps DHIS · [ImageBlock].
+ * @param title controls the text to be shown for the title
+ * @param state Manages the InputShell state
+ * @param supportingText is a list of SupportingTextData that
+ * manages all the messages to be shown
+ * @param legendData manages the legendComponent
+ * @param uploadState controls whether the image is added, loading, or need to be added.
+ * @param downloadButtonVisible controls whether the download button is visible or not
+ * @param isRequired controls whether the field is mandatory or not
+ * @param load to load an image stored in the resource, device memory or from network
+ * we can use loadPainter, loadImageBitmap, loadSvgPainter or loadXmlImageVector
+ * @param painterFor is a composable function which controls how to paint the load param,
+ * @param addButtonText controls the text to be shown for the add button
+ * @param modifier allows a modifier to be passed externally
+ * @param onDownloadButtonClick callback to when download button is clicked
+ * @param onResetButtonClicked callback to when reset button is clicked
+ * @param onAddButtonClicked callback to when add button is clicked
+ */
+@Composable
+internal fun <T> BasicInputImage(
+    title: String,
+    state: InputShellState = InputShellState.UNFOCUSED,
+    supportingText: List<SupportingTextData>? = null,
+    legendData: LegendData? = null,
+    uploadState: UploadState = UploadState.ADD,
+    downloadButtonVisible: Boolean = true,
+    isRequired: Boolean = false,
+    load: suspend () -> T,
+    painterFor: (@Composable (T) -> Painter)? = null,
+    testTag: String = "",
+    addButtonText: String,
+    modifier: Modifier = Modifier,
+    onDownloadButtonClick: () -> Unit,
+    onResetButtonClicked: () -> Unit,
+    onAddButtonClicked: () -> Unit,
+) {
+    InputShell(
+        modifier = modifier.testTag("INPUT_$testTag"),
+        title = title,
+        state = state,
+        isRequiredField = isRequired,
+        legend = {
+            legendData?.let {
+                Legend(legendData, modifier.testTag("INPUT_" + testTag + "_LEGEND"))
+            }
+        },
+        supportingText = {
+            supportingText?.forEach { label ->
+                SupportingText(
+                    label.text,
+                    label.state,
+                    modifier = modifier.testTag("INPUT_" + testTag + "_SUPPORTING_TEXT"),
+                )
+            }
+        },
+        inputField = {
+            when (uploadState) {
+                UploadState.ADD -> {
```

Comment:
this progress indicator, button and custom composable states are shared with more components, should we create other internal component?

Reply:
@DavidAparicioAlbaAsenjo This will make the code hardly coupled with each other, and any change to particular component would impact others as well. If we already have custom components created for buttons and progress indicator, then it's better to implement them individually in each component.